components/pd_client/src/client.rs

```diff
@@ -45,12 +49,56 @@ impl RpcClient {
             v => v.checked_add(1).unwrap_or(std::isize::MAX),
         };
         for i in 0..retries {
-            match validate_endpoints(Arc::clone(&env), cfg, &security_mgr) {
+            match validate_endpoints(Arc::clone(&env), cfg, security_mgr.clone()) {
                 Ok((client, members)) => {
-                    return Ok(RpcClient {
+                    let rpc_client = RpcClient {
                         cluster_id: members.get_header().get_cluster_id(),
-                        leader_client: LeaderClient::new(env, security_mgr, client, members),
-                    });
+                        leader_client: Arc::new(LeaderClient::new(
+                            env,
+                            security_mgr,
+                            client,
+                            members,
+                        )),
+                    };
+
+                    // spawn a background future to update PD information periodically
+                    let duration = cfg.update_interval.0;
+                    let client = Arc::downgrade(&rpc_client.leader_client);
```

Comment:
How about `Arc::clone`?

Reply:
We can not use `Arc::clone` here. If we use `Arc::clone`, the future will hold a strong reference to the client. Because this future is long-lasting, this will prevent the client from being dropped. Holding a weak reference allows the client to be dropped correctly and we can gracefully shutdown the background future when the client is dropped.